plugins/config.go

```diff
@@ -41,47 +42,98 @@ func (pc *registarConfig) RegisterLOOP(loopID string, cmdName string) (func() *e
 	return cmdFn, pc.grpcOpts, nil
 }
 
-// EnvConfig is the configuration interface between the application and the LOOP executable. The values
-// are fully resolved and static and passed via the environment.
+// EnvConfig is the configuration interface between the application and the LOOP executable.
+// The values are fully resolved and static and passed via the environment.
 type EnvConfig interface {
 	PrometheusPort() int
+
+	// Enables trace generation and collection on the plugin
+	TracingEnabled() bool
+
+	// The logical address of the trace collector
+	TracingCollectorTarget() string
+
+	// Attributes to be added to the node's tracing context
+	TracingAttributes() map[string]string
 }
 
 // SetCmdEnvFromConfig sets LOOP-specific vars in the env of the given cmd.
+// It also, due to the plugin library, forwards all env vars from the host automatically.
+// TODO: BCF-2662: Remove once we can use skipHostEnv in hashicorp/go-plugin v1.5.0
+// This method is consumed by the host.
 func SetCmdEnvFromConfig(cmd *exec.Cmd, cfg EnvConfig) {
-	forward := func(name string) {
-		if v, ok := os.LookupEnv(name); ok {
-			cmd.Env = append(cmd.Env, name+"="+v)
-		}
+	injectEnv := map[string]string{
+		"CL_PROMETHEUS_PORT":       strconv.Itoa(cfg.PrometheusPort()),
+		"TRACING_ENABLED":          strconv.FormatBool(cfg.TracingEnabled()),
+		"TRACING_COLLECTOR_TARGET": cfg.TracingCollectorTarget(),
+	}
+
+	for k, v := range cfg.TracingAttributes() {
+		injectEnv["TRACING_ATTRIBUTE_"+k] = v
+	}
+
+	for k, v := range injectEnv {
+		cmd.Env = append(cmd.Env, k+"="+v)
 	}
-	forward("CL_LOG_SQL_MIGRATIONS")
-	cmd.Env = append(cmd.Env,
-		"CL_PROMETHEUS_PORT="+strconv.FormatInt(int64(cfg.PrometheusPort()), 10),
-	)
 }
 
 // GetEnvConfig deserializes LOOP-specific environment variables to an EnvConfig
+// This method is consumed by the plugin.
 func GetEnvConfig() (EnvConfig, error) {
 	promPortStr := os.Getenv("CL_PROMETHEUS_PORT")
 	promPort, err := strconv.Atoi(promPortStr)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse CL_PROMETHEUS_PORT = %q: %w", promPortStr, err)
 	}
 
-	return NewEnvConfig(promPort), nil
+	var tracingEnabled bool
+	tracingEnabledString := os.Getenv("TRACING_ENABLED")
+	if tracingEnabledString == "" {
+		tracingEnabled = false
+	} else {
+		tracingEnabled, err = strconv.ParseBool(tracingEnabledString)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse TRACING_ENABLED = %q: %w", tracingEnabledString, err)
+		}
+	}
+
+	tracingCollectorTarget := os.Getenv("TRACING_COLLECTOR_TARGET")
```

Comment:
is there validation that needs to be done on this if tracing is enabled (at minimum it's not empty?) if so, and we shouldn't bother the parsing in the previous comment, it may be worth pulling out the trace env parsing into a helper function

Reply:
Agreed, I'll do that.